ngraph/test/reshape_opt_kernel.cpp

```diff
@@ -0,0 +1,232 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <numeric>
+#include <vector>
+
+#include "gtest/gtest.h"
+
+#include "ngraph/axis_vector.hpp"
+#include "ngraph/runtime/opt_kernel/reshape.hpp"
+#include "ngraph/shape.hpp"
+
+#include "util/ndarray.hpp"
+
+using namespace ngraph;
+
+namespace
+{
+    using ElementValue = int32_t;
+    enum class AxisOrder
+    {
+        straight,
+        reverse,
+    };
+
+    AxisVector get_axis_order(AxisOrder order, size_t size)
+    {
+        AxisVector v(size);
+        std::iota(begin(v), end(v), 0);
+        if (order == AxisOrder::reverse)
+        {
+            std::reverse(begin(v), end(v));
+        }
+        return v;
+    }
+
+    struct TestParams
+    {
+        AxisOrder order;
+        test::NDArrayBase<ElementValue> input;
+        test::NDArrayBase<ElementValue> output;
+    };
+
+    struct ReshapeOptKernel : ::testing::TestWithParam<TestParams>
+    {
+    };
+
+} // namespace
+
+TEST_P(ReshapeOptKernel, reshape_opt_kernel)
+{
+    const TestParams& p = GetParam();
+
+    const AxisVector axis_order = get_axis_order(p.order, p.input.get_shape().size());
+    std::vector<ElementValue> output_buff(p.input.get_vector().size());
+
+    runtime::opt_kernel::reshape((const char*)p.input.data(),
+                                 (char*)output_buff.data(),
+                                 p.input.get_shape(),
+                                 axis_order,
+                                 p.output.get_shape(),
+                                 sizeof(ElementValue));
+    EXPECT_EQ(p.output.get_vector(), output_buff);
+}
+
+INSTANTIATE_TEST_CASE_P(reshape_opt_kernel,
+                        ReshapeOptKernel,
+                        ::testing::Values(TestParams{AxisOrder::straight,
+                                                     test::NDArray<ElementValue, 2>{
```

Comment:
Do we cover all specific cases? Like without reorder or other root cases

Reply:
Test cover without reorder case `AxisOrder::straight` and reverse order `AxisOrder::reverse` now.

I'm wonder if `AxisOrder` have to validate in ref impl if it has unique values or is this done by ngraph.
Till now other than 'no reorder' and 'reverse order' is not cover.